leapsectz/leapsectz.go

```diff
@@ -20,54 +20,109 @@ limitations under the License.
 package leapsectz
 
 import (
+	"bytes"
 	"encoding/binary"
 	"errors"
 	"io"
+	"io/ioutil"
 	"os"
 	"time"
 )
 
 const file = "/usr/share/zoneinfo/right/UTC"
 
 var errBadData = errors.New("malformed time zone information")
+var errBadVersion = errors.New("version is not supported in file")
 
 // LeapSecond represents a leap second
 type LeapSecond struct {
 	Tleap uint32
-	Nleap uint32
+	Nleap int32
+}
+
+// LeapSecond represents a leap second
+type LeapSecondV2 struct {
+	Tleap uint64
+	Nleap int32
+}
+
+// Header represents file header structure. Fields names are copied from doc
+type Header struct {
+	// A four-octet unsigned integer specifying the number of UTC/local indicators contained in the body.
+	isutccnt uint32
+	// A four-octet unsigned integer specifying the number of standard/wall indicators contained in the body.
+	isstdcnt uint32
+	// A four-octet unsigned integer specifying the number of leap second records contained in the body.
+	leapcnt uint32
+	// A four-octet unsigned integer specifying the number of transition times contained in the body.
+	timecnt uint32
+	// A four-octet unsigned integer specifying the number of local time type Records contained in the body - MUST NOT be zero.
+	typecnt uint32
+	// A four-octet unsigned integer specifying the total number of octets used by the set of time zone designations contained in the body.
+	charcnt uint32
 }
```

Comment:
As in Go the visibility is controlled by capitalization, it normally makes very little sense to have exported structs with only non-exported fields (unless it's intentional and all the useful actions are done via exported functions, which is not the case here).

So I suggest making all fields exported, which will also allow you to use `binary.Read/Write` with this struct. Or making whole Header unexported.

Reply:
Switched to exported fields